index.php

```diff
@@ -64,30 +64,38 @@ public function init() {
 			}
 
 			// Insert filters and hooks here
-			add_action( 'pre_get_posts', array( $this, 'filter_ical_query' ) );
+			add_filter( 'tribe_ical_feed_month_view_query_args', array( $this, 'filter_ical_query' ) );
+			add_filter( 'tribe_events_ical_events_list_args', array( $this, 'filter_ical_query' ) );
 		}
 
 		/**
 		 * Filtering the query for dates
 		 *
-		 * @param $query
+		 * @param $args
+		 *
+		 * @return array
 		 */
-		public function filter_ical_query( $query ) {
+		public function filter_ical_query( $args ) {
 
-			if ( ! isset( $_GET['ical'] )
-			     || ! isset( $query->tribe_is_event_query )
-			     || ! $query->tribe_is_event_query ) {
+			// If ical is not set in the URL then bail
+			if ( ! isset( $_GET['ical'] ) || $_GET['ical'] != 1 ) {
```

Comment:
This too should be sanitized before checking.
Not to waste a check, though, user `filter_input_array`:

```php
$filters = [
    'ical' => FILTER_SANITIZE_INT,
    'tribe_display' => FILTER_SANITIZE_STRING,
    'start_date' => FILTER_SANITIZE_STRING,
    'end_date' => FILTER_SANITIZE_STRING,
];
$vars = filter_input_array( INPUT_GET, $filters );
```

The code can then use the safe `$vars` variable:

```php
if( ! isset( $vars ['ical'] ) { //... }
```

Reply:
For the number it was 
`FILTER_SANITIZE_NUMBER_INT`
but all good. This is nice.